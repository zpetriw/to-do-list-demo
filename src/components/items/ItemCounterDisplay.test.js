import React from 'react';
import expect from 'expect';
import {shallow, render} from 'enzyme';
import ConnectedItemCounterDisplay, {ItemCounterDisplay} from './ItemCounterDisplay';
import {ListItemText, ListItem} from 'material-ui';
import renderer from 'react-test-renderer';

describe('When ItemCounterDisplay renders', () => {

    describe('its bare component', () => {
        
        let wrapper;
        let props;
    
        beforeEach( () => {
            props = {
                // We need to pass in 'classes' manually because it normally gets this from the connected WithStyles() component.
                classes: {counter: "some class"},
                itemCount: 2
            };
            wrapper = shallow(<ItemCounterDisplay {...props}/>);
        });

        it('should render correctly', () => {
            const tree = renderer
                .create(<ItemCounterDisplay {...props}/>)
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('should render <ListItem> and <ListItemText>', () => {
            expect(wrapper.find(ListItem).length).toBe(1);
            expect(wrapper.find(ListItemText).length).toBe(1);
        });
    
        it('should render with correct text', () => {
            expect(wrapper.find(ListItemText).props().primary).toBe("Items To Do: 2");
        });
    
        it('should render with style props', () => {
            expect(wrapper.find(ListItemText).props().className).toBe("some class");
        });
    
    });
    
    describe('its WithStyles() connected component', () => {
    
        let wrapper;
        let props;
    
        beforeEach( () => {
            props = {
                itemCount: 2
            };
            
            wrapper = shallow(<ConnectedItemCounterDisplay {...props}/>);
        });
        // Note on .dive(): 
        // Shallow render the one non-DOM child of the current wrapper, and return a wrapper around the result.
        // This is helpful for testing shallow-rendered connected components. 

        it('should render <ListItem> and <ListItemText>', () => {
            expect(wrapper.dive().find(ListItem).length).toBe(1);
            expect(wrapper.dive().find(ListItemText).length).toBe(1);
        });
        
        it('should render with correct text', () => {
            expect(wrapper.dive().find(ListItemText).props().primary).toBe("Items To Do: 2");
        });

        it('should render a WithStyles() className', () => {
            //expect(wrapper.dive().find(ListItemText).props().className).toEqual(expect.stringContaining("ItemCounterDisplay"));
            expect(wrapper.dive().find(ListItemText).props().className.length).toBeGreaterThan(0);
        });
    });
});
