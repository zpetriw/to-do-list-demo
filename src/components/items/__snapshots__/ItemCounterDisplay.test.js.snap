// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When ItemCounterDisplay renders its bare component should render correctly 1`] = `
<li
  className="MuiListItem-root-1 MuiListItem-default-4 MuiListItem-gutters-8"
  disabled={false}
>
  <div
    className="MuiListItemText-root-11 some class"
  >
    <h3
      className="MuiTypography-root-17 MuiTypography-subheading-24 MuiListItemText-primary-14"
    >
      Items To Do: 2
    </h3>
  </div>
</li>
`;
